src/pudl/metadata/resources/eia.py

```diff
@@ -15,6 +15,30 @@
         "etl_group": "entity_eia",
         "field_namespace": "eia",
     },
+    "boiler_generator_association_types_eia": {
+        "description": "A coding table describing different types of boiler-generator associations in the EIA-860.",
+        "schema": {
+            "fields": ["code", "label", "description"],
+            "primary_key": ["code"],
+            "foreign_key_rules": {"fields": [["boiler_generator_association_code"]]},
```

Comment:
I know it's kind of a long name, but for consistency with the other coding tables I think this column should probably be called `boiler_generator_association_type_code`. IIRC, the existing pattern is:

* column = `(name)_code`
* table = `(name)s_(data_source)`

E.g. the column `energy_source_code` points at the table `energy_sources_eia` and `contract_type_code` points at `contract_types_eia` (and this is the pattern that you followed with the `steam_plant_types_eia` and `steam_plant_type_code` below)

Reply:
that is long but i like it! and that feels right for sure